lib/spack/spack/environment/environment.py

```diff
@@ -2457,15 +2346,141 @@ def make_repo_path(root):
     return path
 
 
-def prepare_config_scope(env):
-    """Add env's scope to the global configuration search path."""
-    for scope in env.config_scopes():
+def included_config_scopes(manifest: "EnvironmentManifestFile") -> List[spack.config.ConfigScope]:
```

Comment:
That's a great point.  Given the nature of the class I wasn't sure if making them methods would be "acceptable".  I've checked and it's only used one place (now?) so I'll make the change.

Reply:
Done.

Reply:
> Is there a reason for all of these functions to be at module scope instead of being EnvironmentManifestfile methods?

(Just leaving a comment as I read the PR) Previously no method in `EnvironmentManifestFile` had side effects on globals, so that could have been a reason to keep a function at module scope for `prepare_*`, `deactivate_*` and `use_*`[^1]

[^1]: The fact is, this function doesn't have only a `manifest` attribute, but also an implicit `configuration` attribute via global variable